sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java

```diff
@@ -3070,8 +3147,8 @@ public Mono<OpenAIFile> getFile(String fileId) {
      * @return represents an assistant that can call the model and use tools on successful completion of {@link Mono}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    Mono<OpenAIFile> uploadFile(FileDetails file, FilePurpose purpose, String filename) {
-        // Generated convenience method for uploadFileWithResponse
+    public Mono<OpenAIFile> uploadFile(FileDetails file, FilePurpose purpose, String filename) {
```

Comment:
`FileDetails` is an artifact of code gen that we have no control over. Since this method sends multipart data and it was decided that according to the general spec the `filename` in the part is not mandatory, but in the case of the OpenAI backend it is, we need to make this field required.